Assignee: Martin He

### Problem 1: Generalized Linear Mixed-Effects Model

In [3]:
## Load Generalized Linear Mixed-Effects Model library
library(lme4)

## Import data from csv file
data <- read.csv("gababies.csv")
head(data)

momid birthord momage timesnc lowbrth bweight delwght lastwght initage
1 39    1        15      0      0       3720    -1240   2480     15     
2 39    2        17      2      0       3260    -1240   2480     15     
3 39    3        19      4      0       3910    -1240   2480     15     
4 39    4        24      9      0       3320    -1240   2480     15     
5 39    5        25     10      1       2480    -1240   2480     15     
6 62    1        17      0      1       2381     -170   2211     17     
  initwght cinitage
1 3720     -2.545  
2 3720     -2.545  
3 3720     -2.545  
4 3720     -2.545  
5 3720     -2.545  
6 2381     -0.545

1. Perform generalized linear mixed-effects model.

In [7]:
model <- glmer(lowbrth ~ initage + timesnc + birthord + (1 | momid),
               data = data, family = binomial(link = "logit"))
summary(model)

Generalized linear mixed model fit by maximum likelihood (Laplace
  Approximation) [glmerMod]
 Family: binomial  ( logit )
Formula: lowbrth ~ initage + timesnc + birthord + (1 | momid)
   Data: data

     AIC      BIC   logLik deviance df.resid 
  1193.0   1217.6   -591.5   1183.0      995 

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.6992 -0.5782 -0.3415  0.6792  2.6610 

Random effects:
 Groups Name        Variance Std.Dev.
 momid  (Intercept) 2.311    1.52    
Number of obs: 1000, groups:  momid, 200

Fixed effects:
            Estimate Std. Error z value Pr(>|z|)   
(Intercept)  1.70425    0.80659   2.113  0.03461 * 
initage     -0.12500    0.04445  -2.812  0.00492 **
timesnc     -0.03025    0.03644  -0.830  0.40649   
birthord    -0.05800    0.09132  -0.635  0.52533   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
         (Intr) initag timsnc
initage  -0.959              
timesnc   0.111 -0.023       
birthord

2. Consider a newborn baby who is a 5th child born to a mother who has a initial birth age of 15 and timesnc of 10. This baby had a birthweight that was considered to be low (i.e. lowbrth == 1). Compute the effect of this mother to give birth to a child with low birthweight.

Given the result of the model above, we may calculate the log odds of the regression model.
* y-intercept = 1.70425
* Coefficient of initage = -0.12500
* Coefficient of timesnc = -0.03025
* Coefficient of birthord = -0.05800
Log odds = 1.70425 + (-0.12500)*15 + (-0.3025)*10 + (-0.05800)*5
= -0.76325
Then, we need to use the logistic function to convert the log odds to a probability:
e^(-0.76325)/(1+e^(-0.76325)) = 0.317941
 
This means that if this newborn baby is a 5th child born to a mother who has an initial birth age of 15 and timesnc of 10, there is about 31.8% chance of baby being born with a low birth weight.

3. The random effect (momid) has a variance of 2.311. Calculate the intraclass correlation coefficient (ICC) for the model, which represents the proportion of the total variance in the odds of giving birth to a child with low birthweight that is attributable to differences between mothers. Use (pie^2)/3 for variance within.

<b>ICC = 2.311 / (pie^2/3 + 2.311) = 0.412615.</b><br>
This means that there's about 41.3% of total variance in the odds of giving birth to a child with a low birth weight attributed to differences between the mothers.